### ISSUE TITLE
[import/no-restricted-paths]: problems with except

### ISSUE THREAD

**Issue by budarin** [import-js/eslint-plugin-import#2818]
The folder structure
<img width="278" alt="image" src="https://github.com/import-js/eslint-plugin-import/assets/8055157/aee6f292-449b-4754-b31b-c86818865032">

I need to prevent modules from `./src/domain/` from importing any modules not from the `./src/domain/` folder, except for any modules from the `./src/utils` folder

```js
        'import/no-restricted-paths': [
            'error',
            {
                zones: [
                    {
                        target: ['./src/domain/**/*'],
                        from: ['./src/!(domain)/**/*'],
                        except: ['./src/utils/**/*'],
                    },
                ],
            },
        ],
```
But the except section does not work - errors are displayed when importing modules from the `./src/utils/` folder

have a look at the [repo](https://github.com/budarin/zustand-clean-architecture)

I nave commented the rule - uncomment it for you

**Comment by kylemh** [comment]
Same behavior even for just the folder

example zone:
```
{
  target: './**/*',
  from: './src/hooks/useClient.ts',
  except: ['./src/hooks/data'],
},
```

Trying to allow `useClient` only to be imported by files in `src/hooks/data`.

**Comment by budarin** [comment]
sorry - my mistake!
except should contain a path relative to from

that's right:
```js
 'import/no-restricted-paths': [
            'error',
            {
                zones: [
                    {
                        target: './src/domain',
                        from: './src',
                        except: ['./domain'],
                    },
                ],
            },
        ],
```
